homeassistant/components/sensor/linky.py

```diff
@@ -0,0 +1,97 @@
+"""
+Support for Linky.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor.linky/
+"""
+import logging
+import json
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, CONF_TIMEOUT
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pylinky==0.1.6']
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=10)
+DEFAULT_TIMEOUT = 10
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Configure the platform and add the Linky sensor."""
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+    timeout = config[CONF_TIMEOUT]
+
+    from pylinky.client import LinkyClient, PyLinkyError
+    client = LinkyClient(username, password, None, timeout)
+
+    try:
+        client.fetch_data()
+    except PyLinkyError as exp:
+        _LOGGER.error(exp)
+        return
+    except BaseException as exp:
```

Comment:
Remove this.

Reply:
Don't know if your read a bit about `pylinky` package, but most of the errors are not handled within `PyLinkyError` error. So it won't be handled.

Reply:
We should only catch specific expected errors, and not use broad except. If there is a specific error that we expect we can add an except block for that.